File: go/darts/darts.go
package darts

import "math"

const outerRadius float64 = 10
const middleRadius float64 = 5
const innerRadius float64 = 1

func Score(x, y float64) int {

	distance := math.Hypot(x, y)

	if distance <= innerRadius {
		return 10
	} else if distance <= middleRadius {
		return 5
	} else if distance <= outerRadius {
		return 1
	} else {
		return 0
	}

}

File: javascript/darts/darts.js
const OUTER_RADIUS = 10;
const MIDDLE_RADIUS = 5;
const INNER_RADIUS = 1;

export const score = (x, y) => {
  const h = Math.hypot(x, y)
  if (h <= INNER_RADIUS) {
    return 10;
  } else if (h <= MIDDLE_RADIUS) {
    return 5;
  } else if (h <= OUTER_RADIUS) {
    return 1;
  } else {
    return 0;
  }
};
